### PR TITLE
Add new table and repo for envelope supplier statements

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-1358

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    @Autowired ObjectMapper objectMapper;
+    @Autowired ClockProvider clockProvider;
+
+    @Test
+    void should_save_and_find_statement() throws Exception {
+        // given
+        var statement =
+            new NewEnvelopeSupplierStatement(
+                LocalDate.now(),
+                "{ \"a\": 100 }",
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:36]: no need for spaces right? not important really - just simple comment here

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Autowired SupplierStatementRepository repo;
+    @Autowired ObjectMapper objectMapper;
+    @Autowired ClockProvider clockProvider;
+
+    @Test
+    void should_save_and_find_statement() throws Exception {
+        // given
+        var statement =
+            new NewEnvelopeSupplierStatement(
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:34]: no need for that new line right? also not important comment

Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:34]: imho more readable

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:36]: readability above all ;)

**Pull request merged**